### ISSUE TITLE
Bug: false positive: "no-unnecessary-qualifier"

### ISSUE THREAD

**Issue by benny-medflyt** [palantir/tslint#4387]
### Bug Report

- __TSLint version__: 5.11.0
- __TypeScript version__: 3.2.2

#### TypeScript code being linted

```ts
export namespace OuterNamespace {
    export const Foo = "Foo";

    export namespace InnerNamespace {
        export function Foo(): any {
            return {
                type: OuterNamespace.Foo
            };
        }
    }
}

console.log(OuterNamespace.InnerNamespace.Foo());
```

with `tslint.json` configuration:

```json
{
    "extends": ["tslint:latest"],
    "rules": {
        "no-namespace": false,
        "no-shadowed-variable": false,
        "no-console": false,
        "no-unnecessary-qualifier": true
    }
}
```

#### Actual behavior

```
Qualifier is unnecessary since 'OuterNamespace' is in scope. (no-unnecessary-qualifier)
   5 |         export function Foo(): any {
   6 |             return {
>  7 |                 type: OuterNamespace.Foo,
     |                      ^
   8 |             };
   9 |         }
  10 |     }

```

#### Expected behavior

This should produce no tslint errors. Note that if we remove the qualifier `OuterNamespace` then the output of the program will change, from:

```
{ type: 'Foo' }
```

to:

```
{ type: [Function: Foo] }
```

**Comment by JoshuaKGoldberg** [comment]
Duplicate of #2413. 

It might also be fixed by the soon-to-be-released 5.12.0 (🎉!); if not, likely still a duplicate.
